youtube_dl/extractor/miomio.py

```diff
@@ -60,10 +61,12 @@ def _real_extract(self, url):
             'http://www.miomio.tv/mioplayer/mioplayerconfigfiles/xml.php?id=%s&r=%s' % (id, random.randint(100, 999)),
             video_id)
 
-        # the following xml contains the actual configuration information on the video file(s)
-        vid_config = self._download_xml(
+        vid_config_request = compat_urllib_request.Request(
             'http://www.miomio.tv/mioplayer/mioplayerconfigfiles/sina.php?{0}'.format(xml_config),
-            video_id)
+            headers={'Referer': 'http://www.miomio.tv/mioplayer/mioplayer-v3.0.swf'})
```

Comment:
```
<embed class="mplaybk" height="520" width="1162" wmode="Opaque" bgcolor="#FFFFFF" allowscriptaccess="always" rel="noreferrer" src="/mioplayer/mioplayer-v3.0.swf" type="application/x-shockwave-flash" flashvars="type=sina&amp;vid=vsq3pwJunwJul&amp;cid=223353" allowfullscreen="true" quality="high">
```
Looks like the SWF URL can be retrieved from the webpage. It's better to extract the SWF URL from it and fallback to the hardcoded one instead of using a hardcoded one directly. An similar case is what [TudouIE](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/tudou.py#L57-L59) does. Doing so prevents future failures from SWF URL changes.